portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -53,11 +53,13 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    int limit = Integer.parseInt(request.getParameter("limit"));
     Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     PreparedQuery results = datastore.prepare(query);
     ImmutableList<String> comments = 
                             Streams.stream(results.asIterable())
+                            .limit(limit)
                             .map(entity -> entity.getProperty("content").toString())
                             .collect(toImmutableList());
 
```

Comment:
super nit: while we're here, could we fix the formatting real quick? These lines should just be 4 spaces in from the beginning of "ImmutableList" (so aligned with the "t")